### PR TITLE
IDEA-231570 Support new LogBuilder API from Log4J 2

### PR DESCRIPTION
**Pull request by MariusVolkhart**
Modify logging inspections to support for Log4J 2's new [LogBuilder API](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/index.html?org/apache/logging/log4j/LogBuilder.html).

In short, `LOGGER.info("My name is {}", "John")` can now be written as `LOGGER.atInfo().log("My name is {}", "John")`. The inspections will detect problems in the new `log(...)` methods now.

### CODE REVIEW

File: plugins/InspectionGadgets/testsrc/com/siyeh/ig/logging/PlaceholderCountMatchesArgumentCountInspectionTest.java [hunk 1]
```diff
@@ -35,6 +38,14 @@ protected LocalInspectionTool getInspection() {
       "package org.apache.logging.log4j.util;" +
       "public interface Supplier<T> {" +
       "    T get();" +
+      "}",
+
+      "package org.apache.logging.log4j;" +
+      "import org.apache.logging.log4j.util.Supplier;" +
+      "public interface LogBuilder {" +
+      "  public void log(String format, Object p0);" +
```
Comment (MariusVolkhart) [inline plugins/InspectionGadgets/testsrc/com/siyeh/ig/logging/PlaceholderCountMatchesArgumentCountInspectionTest.java]: This overload is not yet in the released version, but has been merged and will be in the the next version. See https://github.com/apache/logging-log4j2/commit/6913465c030ce02b3d6abb5a11b31f5f392f6b2a

**Comment by ybedrov** [comment]
Hello,
Have you submitted a contributor license agreement? If not, please follow the steps described at http://www.jetbrains.com/agreements/cla/ to sign it.

**Comment by MariusVolkhart** [comment]
Yes, I've submitted the CLA

**Comment by BasLeijdekkers** [comment]
Merged manually. Thank you for your contribution!

**Pull request closed without merging**